packages/cozy-viewer/src/Panel/AI/AIAssistantPanel.jsx

```diff
@@ -49,25 +49,36 @@ const AIAssistantPanel = ({ className }) => {
     }
   }
 
+  const extractFileContent = async (client, fileBlob, file) => {
+    if (isTextFile(file.mime)) {
+      return await fileBlob.text()
+    }
+
+    const rawTextContent = await extractText(client, fileBlob, {
+      name: file.name,
+      mime: file.mime
+    })
+    return rawTextContent ? JSON.stringify(rawTextContent) : ''
+  }
+
+  const validateContentSize = textContent => {
+    const summaryConfig = flag('drive.summary')
+    if (
+      summaryConfig?.maxTokens &&
+      roughTokensEstimation(textContent) > summaryConfig.maxTokens
+    ) {
+      const error = new Error('DOCUMENT_TOO_LARGE')
+      error.code = 'DOCUMENT_TOO_LARGE'
+      throw error
+    }
+  }
```

Comment:
You should move these functions outside the component if they do not use state (maybe in same helpers?)